mungegithub/mungers/approvers/owners.go

```diff
@@ -137,15 +137,46 @@ func findMostCoveringApprover(allApprovers []string, reverseMap map[string]sets.
 	return bestPerson
 }
 
+// temporaryUnapprovedFiles returns the list of files that wouldn't be
+// approved by the given set of approvers.
+func (o Owners) temporaryUnapprovedFiles(approvers sets.String) sets.String {
+	ap := NewApprovers(o)
+	for approver := range approvers {
+		ap.AddApprover(approver, "", "")
+	}
+	return ap.UnapprovedFiles()
+}
+
+// KeepMostCoveringApprovers finds who we should keep as suggested approvers given a pre-selection
+// knownApprovers must be a subset of potentialApprovers.
+func (o Owners) KeepMostCoveringApprovers(knownApprovers sets.String, potentialApprovers []string) sets.String {
+	keptApprovers := sets.NewString()
+
+	unapproved := o.temporaryUnapprovedFiles(knownApprovers)
+	reverseMap := o.GetReverseMap()
+
+	for _, suggestedApprover := range o.GetSuggestedApprovers(potentialApprovers).List() {
+		if reverseMap[suggestedApprover].Intersection(unapproved).Len() != 0 {
```

Comment:
there is no "max" computation.  how can it be Keep**Most**CoveringApprovers?  Of the files that are unapproved using a given set of people, you want to choose the the person that covers the largest set right?  Unless you just want to choose randomly the people that can cover the rest